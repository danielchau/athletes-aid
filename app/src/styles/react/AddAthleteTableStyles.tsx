import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Add Athletes Table.
 */
export const addAthleteTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px",
            height: "100%",
            display: "flex",
            justifyContent: "center"
        },
        tableContainer: {
            width: "100%",
            height: "100%",
            padding: "8px",
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            overflow: "scroll"
        },
        table: {
            width: "100%"
        },
        tableBodyContainer: {
            width: "2000px",
            overflow: "auto",
            height: "100%"
        },
        tableBody: {
            tableLayout: "fixed"
        },
        tableRow: {
            cursor: "pointer"
        },
        tooltip: {
            backgroundColor: "#fff",
            color: "#4f4f4f",
            boxShadow: theme.shadows[1]
        }
    })
);
