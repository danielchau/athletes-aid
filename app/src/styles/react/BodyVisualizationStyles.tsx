import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Body Visualization.
 */
export const bodyVisualizationStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            height: "100%"
        },
        imageContainer: {
            position: "relative",
            alignSelf: "center"
        },
        image: {
            width: "100%"
        },
        dot: {
            height: "15px",
            width: "15px",
            position: "absolute",
            borderRadius: "15px"
        },
        legendContainer: {
            borderLeft: "#dedede solid 1px"
        },
        legendItem: {
            display: "flex",
            color: "rgba(0, 0, 0, 0.54)",
            marginBottom: "2px"
        },
        dotLegend: {
            height: "10px",
            width: "10px",
            borderRadius: "10px",
            marginTop: "6px",
            marginRight: "4px"
        }
    })
);
