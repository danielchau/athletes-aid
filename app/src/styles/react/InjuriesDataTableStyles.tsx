import { makeStyles, Theme, createStyles, lighten } from "@material-ui/core/styles";

/**
 * React styles for the Injury Data Table.
 */
export const injuriesDataTableStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            marginTop: theme.spacing(3)
        },
        paper: {
            width: "100%",
            marginBottom: theme.spacing(2),
            backgroundColor: "rgba(0, 33, 69, 0.05)"
        },
        table: {
            minWidth: 750
        },
        tableWrapper: {
            overflowX: "auto"
        },
        tableRow: {
            cursor: "pointer"
        },
        visuallyHidden: {
            border: 0,
            clip: "rect(0 0 0 0)",
            height: 1,
            margin: -1,
            overflow: "hidden",
            padding: 0,
            position: "absolute",
            top: 20,
            width: 1
        }
    })
);

/**
 * React styles for the Injury Data Table Toolbar Header.
 */
export const injuriesDataTableToolbarStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            paddingLeft: theme.spacing(2),
            paddingRight: theme.spacing(1)
        },
        highlight:
            theme.palette.type === "light"
                ? {
                      color: theme.palette.primary.main,
                      backgroundColor: lighten(theme.palette.primary.light, 0.85)
                  }
                : {
                      color: theme.palette.text.primary,
                      backgroundColor: theme.palette.secondary.dark
                  },
        title: {
            flex: "1 1 100%"
        },
        export: {
            margin: "auto",
            color: theme.palette.secondary.main,
            display: "flex"
        }
    })
);
