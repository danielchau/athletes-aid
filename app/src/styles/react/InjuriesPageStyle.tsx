import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

/**
 * React styles for the Injuries Page.
 */
export const injuriesPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            padding: "16px",
            backgroundColor: "#fff"
        },
        drawerOpen: {
            width: "calc(100vw - 240px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.enteringScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        drawerClosed: {
            width: "calc(100vw - 72px)",
            transition: theme.transitions.create("width", {
                easing: theme.transitions.easing.sharp,
                duration: theme.transitions.duration.leavingScreen
            }),
            [theme.breakpoints.down("xs")]: {
                width: "100vw"
            }
        },
        grid: {
            backgroundColor: "#fff"
        },
        paper: {
            padding: theme.spacing(2),
            textAlign: "center",
            color: theme.palette.text.secondary,
            backgroundColor: "rgba(0, 33, 69, 0.05)",
            height: "100%"
        },
        vizPaper: {
            padding: theme.spacing(2),
            height: "100%"
        },
        primaryStatisticContainer: {
            display: "flex",
            flexDirection: "column"
        },
        primaryStatisticValue: {
            overflow: "hidden",
            textOverflow: "ellipsis",
            fontSize: "calc((100vw - 240px)/16)"
        },
        primaryStatisticLabel: {
            paddingTop: "16px"
        },
        dateTimeContainer: {},
        textField: {
            marginLeft: theme.spacing(1),
            marginRight: theme.spacing(1),
            width: 200
        },
        dateField: {
            width: 200,
            marginRight: theme.spacing(1),
            marginTop: 0,
            marginBottom: 0
        },
        dateTimeButton: {
            marginLeft: "8px",
            marginTop: "6px"
        }
    })
);
