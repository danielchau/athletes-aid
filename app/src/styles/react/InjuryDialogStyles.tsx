import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryDialogStyles = makeStyles((theme: Theme) =>
    createStyles({
        appBar: {
            position: "relative"
        },
        title: {
            marginLeft: theme.spacing(2),
            flex: 1
        },
        dialogTitle: {
            padding: "0"
        },
        dialogPaper: {
            height: "100%",
            maxWidth: "600px",
            width: "50%",
            minWidth: "200px",
            position: "fixed",
            bottom: "0",
            right: "0",
            margin: "0"
        },
        dialogContentContainer: {
            padding: "0"
        },
        dialogContent: {
            padding: "16px"
        },
        dialogContentPaper: {
            backgroundColor: "#fafafa",
            padding: "8px",
            marginBottom: "8px"
        },
        notePaper: {
            backgroundColor: "#ffffff",
            padding: "8px",
            marginBottom: "8px"
        },
        noteContainer: {
            display: "flex",
            flexDirection: "row"
        },
        noteIcon: {
            margin: "auto",
            marginLeft: "8px",
            marginRight: "16px"
        },
        newNoteContainer: {
            display: "flex",
            flexDirection: "row",
            width: "100%",
            justifyContent: "center",
            alignItems: "center",
            alignSelf: "center"
        },
        newNote: {
            flexGrow: 1,
            paddingRight: "8px"
        },
        sendButton: {
            color: "#F2A71E"
        },
        activeSwitchContainer: {
            display: "flex",
            flexDirection: "row",
            justifyContent: "center",
            alignContent: "center",
            alignSelf: "center",
            paddingBottom: "8px"
        }
    })
);
