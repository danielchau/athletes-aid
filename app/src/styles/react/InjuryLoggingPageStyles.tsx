import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingPageStyles = makeStyles((theme: Theme) =>
    createStyles({
        root: {
            width: "100%",
            padding: "8px"
        },
        backButton: {
            marginRight: theme.spacing(1)
        },
        completedContainer: {
            height: "calc(100% - 116px)",
            textAlign: "center",
            paddingTop: "20%"
        },
        instructions: {
            marginTop: theme.spacing(1),
            marginBottom: theme.spacing(1)
        },
        paper: {
            color: theme.palette.text.secondary,
            backgroundColor: "#fafafa"
        },
        paperContent: {
            height: "100%",
            padding: "16px",
            paddingRight: "8px",
            overflow: "auto"
        },
        stepper: {
            backgroundColor: "#fafafa",
            borderRadius: "4px"
        },
        loggingContent: {
            height: "calc(100% - 168px)",
            marginBottom: "52px",
            marginTop: "8px",
            overflow: "hidden"
        },
        loggingBottomButtons: {
            position: "fixed",
            bottom: 0,
            right: 0,
            height: "52px",
            padding: "8px"
        },
        multilineInput: {
            width: "100%"
        }
    })
);
