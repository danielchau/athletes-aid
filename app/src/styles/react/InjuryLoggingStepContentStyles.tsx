import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";

export const injuryLoggingStepContentStyles = makeStyles((theme: Theme) =>
    createStyles({
        formControl: {
            minWidth: 200,
            width: "50%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        textInput: {
            minWidth: 200,
            width: "100%",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        multilineInput: {
            width: "100%",
            paddingBottom: "16px",
            margin: "0px"
        },
        dateInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "16px"
        },
        checkboxInput: {
            minWidth: 200,
            width: "50%",
            margin: "0px",
            paddingRight: "8px",
            paddingBottom: "23px",
            paddingTop: "7px"
        },
        instructions: {
            textAlign: "center",
            marginBottom: theme.spacing(3),
            color: theme.palette.text.secondary,
            width: "100%"
        }
    })
);
