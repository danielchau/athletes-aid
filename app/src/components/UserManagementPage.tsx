import React from "react";
import { rosterPageStyles } from "../styles/react/RosterPageStyles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import { User, Team } from "../util/types";
import FetchingScreen from "./FetchingScreen";
import {
    TrainerPermissions,
    CoachPermissions,
    AdminPermissions,
    UserPermissions
} from "../util/permissions";
import {
    FormControl,
    Select,
    MenuItem,
    TextField,
    Tooltip,
    IconButton,
    Input
} from "@material-ui/core";
import HelpIcon from "@material-ui/icons/Help";
import HelpDialog from "./HelpDialog";
import { userManagementPageName } from "../constants/constants";
import Chip from "@material-ui/core/Chip";
import {
    changeRoleForUser,
    changeTeamsForUser,
    getAllUsers,
    deleteUser
} from "../actions/UserAction";
import DeleteIcon from "@material-ui/icons/Delete";
import AddIcon from "@material-ui/icons/Add";
import AddUserDialog from "./AddUserDialog";
import ErrorDialog from "./ErrorDialog";
import { fetchTeamsEndpoint } from "../actions/TeamAction";

interface UserManagementPageProps {
    currentUser: User;
}

/**
 * Roster Page dispalys all of the athletes on the selected team and provides navigation to the
 * athletes profiles.
 * @param props
 */
export default function UserManagementPage(props: UserManagementPageProps) {
    const classes = rosterPageStyles({});
    const [users, setUsers] = React.useState<User[]>([]);
    const [isFirstRender, setIsFirstRender] = React.useState<boolean>(true);
    const [isFetching, setIsFetching] = React.useState<boolean>(true);
    const [autocompleteVal, setAutocompleteVal] = React.useState<string>("");
    const [allUsers, setAllUsers] = React.useState<User[]>([]);
    const [open, setOpen] = React.useState(false);
    const [openAddUser, setOpenAddUser] = React.useState(false);
    const [openError, setOpenError] = React.useState(false);
    const [teams, setTeams] = React.useState([]);

    React.useEffect(() => {
        if (isFirstRender) {
            fetchTeamsEndpoint(props.currentUser.permissions).then((t: Team[]) => {
                setTeams(t);
                getAllUsers().then((users: User[] | null) => {
                    if (!!users) {
                        setAllUsers(users);
                        setUsers(users);
                    } else {
                        setOpenError(true);
                    }
                    setIsFirstRender(false);
                    setIsFetching(false);
                });
            });
        }
    }, []);

    React.useEffect(() => {
        console.log(users);
        console.log(allUsers);
    }, [users, allUsers]);

    const handleAutocompleteChange = (event: React.ChangeEvent<{ value: string }>) => {
        let value = event.target.value;

        setAutocompleteVal(value);

        var reg = new RegExp(
            value
                .split("")
                .join("\\w*")
                .replace(/\W/, ""),
            "i"
        );
        let newUsers = allUsers.filter((a: User) => {
            let athleteName = a.firstName + " " + a.lastName;
            if (athleteName.match(reg)) {
                return a;
            }
        });

        setUsers(newUsers);
    };

    const onSelectChange = (event: React.ChangeEvent<{ value: string }>, user: User) => {
        changeRoleForUser(user.cwl, event.target.value).then((role: string | null) => {
            if (!!role) {
                let tempUsers = allUsers;
                tempUsers = tempUsers.map(u => {
                    if (user.cwl == u.cwl) {
                        switch (role) {
                            case "admin":
                                u.permissions = AdminPermissions;
                                break;
                            case "trainer":
                                u.permissions = TrainerPermissions;
                                break;
                            default:
                                u.permissions = CoachPermissions;
                                break;
                        }
                    }
                    return u;
                });
                setUsers(tempUsers);
                setAllUsers(tempUsers);
            } else {
                setOpenError(true);
            }
        });
    };

    const onTeamChange = (event: React.ChangeEvent<{ value: string[] }>, user: User) => {
        let teamsMap = new Map();
        teams.forEach(t => teamsMap.set(t.name + " - " + t.season, t.id));

        changeTeamsForUser(
            user.cwl,
            event.target.value.filter(t => !!t).map(t => teamsMap.get(t))
        ).then((teams: string[] | null) => {
            if (!!teams) {
                let tempUsers = allUsers;
                tempUsers = tempUsers.map(u => {
                    if (user.cwl == u.cwl) u.teams = teams;
                    return u;
                });
                setUsers(tempUsers);
                setAllUsers(tempUsers);
            } else {
                setOpenError(true);
            }
        });
    };

    const onDeleteUser = (user: User) => {
        deleteUser(user.cwl).then((response: string) => {
            if (!!response) {
                let tempUsers = allUsers.slice();
                let deleteIndex = null;
                for (let i = 0; i < tempUsers.length; i++) {
                    if (tempUsers[i].cwl == user.cwl) {
                        deleteIndex = i;
                    }
                }
                if (!!deleteIndex) {
                    tempUsers.splice(deleteIndex);
                }
                setUsers(tempUsers);
                setAllUsers(tempUsers);
            } else {
                setOpenError(true);
            }
        });
    };

    const getUserTeamNames = (t: string[]) => {
        let teamMap = new Map();
        teams.forEach(t => teamMap.set(t.id, t.name + " - " + t.season));
        return t.map(t => teamMap.get(t));
    };

    return (
        <div className={classes.root}>
            <ErrorDialog open={openError} setOpen={setOpenError} />
            <div className={classes.searchBarContainer}>
                <TextField
                    label="Search User..."
                    variant="outlined"
                    fullWidth
                    value={autocompleteVal}
                    onChange={handleAutocompleteChange}
                    className={classes.searchBar}
                />
                <Tooltip title="Add User">
                    <IconButton
                        onClick={() => setOpenAddUser(true)}
                        style={{
                            marginLeft: "8px",
                            marginBottom: "16px",
                            height: "60px",
                            width: "60px"
                        }}
                    >
                        <AddIcon />
                    </IconButton>
                </Tooltip>
                <AddUserDialog
                    open={openAddUser}
                    setOpen={setOpenAddUser}
                    allUsers={allUsers}
                    setAllUsers={setAllUsers}
                    setOpenError={setOpenError}
                    setUsers={setUsers}
                />
            </div>
            {isFetching ? (
                <FetchingScreen />
            ) : (
                <Paper className={classes.tableContainer}>
                    <Tooltip title="Help">
                        <IconButton
                            style={{ position: "absolute", top: "68px", right: "4px" }}
                            onClick={() => setOpen(true)}
                        >
                            <HelpIcon />
                        </IconButton>
                    </Tooltip>
                    <HelpDialog open={open} setOpen={setOpen} page={userManagementPageName} />
                    <div className={classes.tableBodyContainer}>
                        <Table stickyHeader className={classes.tableBody}>
                            <TableHead>
                                <TableRow>
                                    <TableCell>
                                        <b>CWL / Name</b>
                                    </TableCell>
                                    <TableCell align="right">
                                        <b>Teams</b>
                                    </TableCell>
                                    <TableCell align="right">
                                        <b>Role</b>
                                    </TableCell>
                                    <TableCell align="right" style={{ width: "75px" }}></TableCell>
                                </TableRow>
                            </TableHead>
                            <TableBody>
                                {users.map(row => (
                                    <TableRow className={classes.tableRow} key={row.cwl}>
                                        <TableCell component="th" scope="row">
                                            {row.cwl + " / " + row.firstName + " " + row.lastName}
                                        </TableCell>
                                        <TableCell align="right" style={{ padding: "0px" }}>
                                            <FormControl
                                                variant="outlined"
                                                style={{
                                                    width: "300px",
                                                    textAlign: "left"
                                                }}
                                            >
                                                <Select
                                                    id="team-chips"
                                                    multiple
                                                    value={getUserTeamNames(row.teams)}
                                                    onChange={(
                                                        evt: React.ChangeEvent<{ value: string[] }>
                                                    ) => onTeamChange(evt, row)}
                                                    input={<Input id="select-multiple-chip" />}
                                                    renderValue={selected => (
                                                        <div className={classes.chips}>
                                                            {(selected as string[]).map(value => (
                                                                <Chip
                                                                    key={value}
                                                                    label={value}
                                                                    className={classes.chip}
                                                                />
                                                            ))}
                                                        </div>
                                                    )}
                                                >
                                                    {teams.map(team => (
                                                        <MenuItem
                                                            key={team.id}
                                                            value={team.name + " - " + team.season}
                                                        >
                                                            {team.name + " - " + team.season}
                                                        </MenuItem>
                                                    ))}
                                                </Select>
                                            </FormControl>
                                        </TableCell>
                                        <TableCell align="right" style={{ padding: "0px" }}>
                                            <FormControl
                                                variant="outlined"
                                                style={{
                                                    width: "150px",
                                                    textAlign: "left"
                                                }}
                                            >
                                                <Select
                                                    id={row.firstName + " " + row.lastName}
                                                    value={row.permissions.label}
                                                    classes={{
                                                        selectMenu: classes.dropdownMenu
                                                    }}
                                                    onChange={(
                                                        evt: React.ChangeEvent<{ value: string }>
                                                    ) => onSelectChange(evt, row)}
                                                    disabled={row.cwl == props.currentUser.cwl}
                                                >
                                                    {[
                                                        AdminPermissions,
                                                        TrainerPermissions,
                                                        CoachPermissions
                                                    ].map((perm: UserPermissions, i: number) => (
                                                        <MenuItem key={i} value={perm.label}>
                                                            {perm.label}
                                                        </MenuItem>
                                                    ))}
                                                </Select>
                                            </FormControl>
                                        </TableCell>
                                        <TableCell align="right">
                                            {row.cwl != props.currentUser.cwl && (
                                                <Tooltip title="Delete User">
                                                    <IconButton
                                                        style={{ color: "#a83232" }}
                                                        onClick={() => onDeleteUser(row)}
                                                    >
                                                        <DeleteIcon />
                                                    </IconButton>
                                                </Tooltip>
                                            )}
                                        </TableCell>
                                    </TableRow>
                                ))}
                            </TableBody>
                        </Table>
                    </div>
                </Paper>
            )}
        </div>
    );
}
