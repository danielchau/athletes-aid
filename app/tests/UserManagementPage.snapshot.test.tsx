import * as React from 'react'
import UserManagementPage from '../src/components/UserManagementPage'
import renderer from 'react-test-renderer'

const mockUser = {
    athleteProfile: {},
    permissions: {}
}
it('UserManagementPage renders correctly', () => {
  const tree = renderer.create(<UserManagementPage user={mockUser} />).toJSON()
  expect(tree).toMatchSnapshot()
})