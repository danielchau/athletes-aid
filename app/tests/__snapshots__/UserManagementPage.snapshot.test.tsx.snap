// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserManagementPage renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-searchBar-3 MuiFormControl-fullWidth"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
      data-shrink={false}
    >
      Search User...
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value=""
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-67 MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="PrivateNotchedOutline-legendLabelled-69"
        >
          <span>
            Search User...
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
